src/commands/WhoisCommand.ts

```diff
@@ -0,0 +1,93 @@
+import { ChannelLogsQueryOptions, Message, MessageEmbed, TextChannel } from 'discord.js';
+import BotConfig from '../BotConfig';
+import DiscordUtil from '../util/DiscordUtil';
+import PrefixCommand from './PrefixCommand';
+import Command from './Command';
+
+export default class WhoisCommand extends PrefixCommand {
+	public readonly aliases = ['who', 'whois'];
+
+	public async run( message: Message, args: string ): Promise<boolean> {
+		if ( !args.length ) {
+			return false;
+		}
+
+		let fromDiscordWhois = false;
+		if ( args.startsWith( '<@' ) ) {
+			fromDiscordWhois = true;
+		}
+
+		if ( message.deletable ) {
+			try {
+				await message.delete();
+			} catch ( error ) {
+				Command.logger.error( error );
+			}
+		} else {
+			Command.logger.log( 'Message not deletable' );
+		}
+
+
+		const logChannel = await DiscordUtil.getChannel( BotConfig.verification.verificationLogChannel );
+		const allMessages: Message[] = [];
+		let lastId: string | undefined;
+		let continueSearch = true;
+
+		try {
+			while ( continueSearch ) {
```

Comment:
You're going to want to move this to `MojiraBot.ts` because we don't want to be fetching all messages every time the command is run. All we need to do is fetch these on startup and read from `logChannel.messages.cache`.

Reply:
That is a very good point, thank you!